protocols/tls/src/certificate.rs

```diff
@@ -33,6 +33,12 @@ use log::error;
 const LIBP2P_OID: &[u64] = &[1, 3, 6, 1, 4, 1, 53594, 1, 1];
 const LIBP2P_SIGNATURE_ALGORITHM_PUBLIC_KEY_LENGTH: usize = 65;
 static LIBP2P_SIGNATURE_ALGORITHM: &rcgen::SignatureAlgorithm = &rcgen::PKCS_ECDSA_P256_SHA256;
+// preferred, but not supported by rustls yet
+//const LIBP2P_SIGNATURE_ALGORITHM_PUBLIC_KEY_LENGTH: usize = 32;
+//static LIBP2P_SIGNATURE_ALGORITHM: &rcgen::SignatureAlgorithm = &rcgen::PKCS_ED25519;
```

Comment:
It should handle the Ed25519 signature exactly like TLS 1.3 handles any other Ed25519 signature, right?  It's only that quin/rustls must invoke a call back for certificate validation, right?

Reply:
rustls does not support ed25519 ― that’s <https://github.com/ctz/rustls/issues/52>.  That said, the names are misleading: these are X.509 signature algorithms, not libp2p ones.